src/esp/sensor/SensorFactory.h

```diff
@@ -0,0 +1,16 @@
+#include "esp/scene/SceneNode.h"
+#include "esp/sensor/CameraSensor.h"
+#include "esp/sensor/Sensor.h"
+
+namespace esp {
+class SensorFactory {
+ public:
+  virtual ~SensorFactory();
```

Comment:
Why declare a destructor for this class? Probably not going to create an object of this class ever.